pkg/git/executor/executor.go

```diff
@@ -34,8 +33,8 @@ func (e ShellExecutor) Exec(command string, dir string, args ...string) error {
 	ctx, ctxCancel := context.WithTimeout(e.parentCtx, e.timeout)
 	defer ctxCancel()
 	cmd := exec.CommandContext(ctx, command, args...)
-	if e.privateSSHKeyPath != "" {
-		cmd.Env = []string{fmt.Sprintf("GIT_SSH_COMMAND=ssh -i %s", e.privateSSHKeyPath)}
+	if len(e.envVars) > 0 {
+		cmd.Env = e.envVars
```

Comment:
doesn't this override any other env vars we have? Should we first get os.Env()?

Reply:
It does, but we were also overriding them before, and it was actually what I wanted to avoid inheriting a "polluted" shell/Git config.